statsmodels/tsa/statespace/tests/test_mlemodel.py

```diff
@@ -26,44 +36,148 @@ def get_dummy_mod():
 
     mod = sarimax.SARIMAX(endog, exog=exog, order=(0,0,0), time_varying_regression=True, mle_regression=False)
 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        # Regression test to make sure bfgs_tune works
-        res = mod.fit(disp=-1)
+    if fit:
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            res = mod.fit(disp=-1)
+    else:
+        res = None
     
     return mod, res
 
+
 def test_fit_misc():
-    mod, res = get_dummy_mod()
+    true = results_sarimax.wpi1_stationary
+    endog = np.diff(true['data'])[1:]
+
+    mod = sarimax.SARIMAX(endog, order=(1,0,1), trend='c')
 
-    # Test bfgs_tune=True
+    # Test optim_hessian={'opg','oim','cs'}
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        # Regression test to make sure bfgs_tune works
-        res = mod.fit(disp=-1, bfgs_tune=True)
-    assert_almost_equal(res.params, [0,0], 7)
+        res1 = mod.fit(method='ncg', disp=True, optim_hessian='opg')
+        res2 = mod.fit(method='ncg', disp=True, optim_hessian='oim')
+        res3 = mod.fit(method='ncg', disp=True, optim_hessian='cs')
+        assert_raises(NotImplementedError, mod.fit, method='ncg', disp=False, optim_hessian='a')
+    # Check that the Hessians broadly result in the same optimum
+    assert_allclose(res1.llf, res2.llf, rtol=1e-2)
+    assert_allclose(res1.llf, res3.llf, rtol=1e-2)
 
     # Test return_params=True
+    mod, _ = get_dummy_mod(fit=False)
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        # Regression test to make sure bfgs_tune works
         res_params = mod.fit(disp=-1, return_params=True)
 
     assert_almost_equal(res_params, [0,0], 7)
 
+
 def test_score_misc():
     mod, res = get_dummy_mod()
 
     # Test that the score function works
     mod.score(res.params)
 
+
 def test_from_formula():
     assert_raises(NotImplementedError, lambda: MLEModel.from_formula(1,2,3))
 
+
 def test_cov_params():
     mod, res = get_dummy_mod()
 
-    assert_almost_equal(res.cov_params(), res.cov_params_delta)
+    # Smoke test for each of the covariance types
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        res = mod.fit(res.params, disp=-1, cov_type='cs')
+        assert(res.cov_type == 'cs')
```

Comment:
this should not be a python `assert` in unit tests. Either use numpy's `assert_equal` (should be more informative in this case) or `assert_` (trailing underline)

(Main traditional reason, python `assert` are removed when byte compiling python with `-o` flag.)